src/components/Reactions/Reaction.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import {Button, ButtonSize, PaletteOption, PopoverProps, Popup} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import {useReactionsContext} from './context';
+import {useReactionsPopup} from './hooks';
+
+export type ReactionProps = Pick<PaletteOption, 'value' | 'content' | 'title'>;
+
+export interface ReactionStateProps {
+    /**
+     * Reaction's unique value (ID).
+     */
+    value: string;
+    /**
+     * Should be true when the user used this reaction.
+     */
+    selected?: boolean;
+    /**
+     * Display a number after the icon.
+     * Represents the number of users who used this reaction.
+     */
+    counter?: React.ReactNode;
+    /**
+     * If present, when a user hovers over the reaction, a popover appears with `tooltip.content`.
+     * Can be used to display users who used this reaction.
+     */
+    tooltip?: React.ReactNode;
+}
+
+interface ReactionInnerProps extends Pick<PaletteOption, 'content'> {
+    reaction: ReactionStateProps;
+    size: ButtonSize;
+    tooltipBehavior: 'tooltip' | 'popover';
+    onClick?: (value: string) => void;
+}
+
+const popupDefaultPlacement: PopoverProps['placement'] = [
+    'bottom-start',
+    'bottom',
+    'bottom-end',
+    'top-start',
+    'top',
+    'top-end',
+];
+
+const b = block('reactions');
+
+export function Reaction(props: ReactionInnerProps) {
+    const {value, selected, counter, tooltip} = props.reaction;
+    const {size, content, onClick} = props;
+
+    const onClickCallback = React.useCallback(() => onClick?.(value), [onClick, value]);
+
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+    const {onMouseEnter, onMouseLeave} = useReactionsPopup(props.reaction, buttonRef);
+    const {openedTooltip: currentHoveredReaction} = useReactionsContext();
+
+    const button = (
+        <span
+            onMouseEnter={props.tooltipBehavior === 'tooltip' ? onMouseEnter : undefined}
```

Comment:
We can assing handlers directly to the button, no need to add wrapper

Reply:
fixed